### ISSUE TITLE
[Question] "lang not in mapping" behavior: Is data skipped? & Best practices for fine-grained dialects (16 tribes vs 42 dialects)

### ISSUE THREAD

**Issue by Zayn1029** [facebookresearch/omnilingual-asr#69]
## Description
I am fine-tuning the `omniASR_LLM_300M` model on a custom dataset containing low-resource Formosan languages (e.g., Kanakanavu `xnb`, Puyuma `pyu`). These language codes are not present in the pre-trained tokenizer vocabulary.

## Observed Behavior
During training, I see continuous warnings in the logs like this:

```text
lang not in mapping: xnb_Latn
lang not in mapping: pyu_Latn
lang not in mapping: ssf_Latn
...
However, the training process does not crash. The step counter continues to increase, and the loss metrics are updating.

Questions
1. Data Handling Behavior
When lang not in mapping occurs, does the data loader drop/skip these specific samples entirely? I am concerned that my model is currently only training on the subset of supported languages (those that didn't trigger the warning) while silently ignoring the unsupported ones.

2. Best Practice for Unsupported Languages
Since I need to support these specific dialects/languages that are not in the base model's vocabulary, what is the recommended workflow?

**Comment by artemru** [comment]
right, probably, we dont handle this edge case well.
you should extend the lang lookup table with new langs you want to include there and next change the lang emb layer accordingly.
